src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -1195,4 +1167,48 @@ $dark-focusColor: var(--green-500, color(green-500));
   display: inline-flex;
   justify-content: center;
 }
+
+// TODO: KButton theming should live in KButton.vue.
+// TODO: Fix these styles not always providing a solid background color for the copy button allowing content to clip through it.
+.k-code-block-secondary-actions .k-button:not(.increase-specificity) {
```

Comment:
**Comment**: I’ve moved these styles out of the scoped style block so they affect the content put into the secondary-actions slot (which is likely going to be `KButton` instances).

Reply:
suggestion: for readability, and easy to verify all unscoped styles are properly scoped could you put ALL styles in the unscoped block under `.k-code-block`?

ex.

```
<style lang="scss">
@import '@/styles/variables';
@import '@/styles/functions';

.k-code-block {
 // ... all styles
}
</style>
```

Reply:
> suggestion: for readability, and easy to verify all unscoped styles are properly scoped could you put ALL styles in the unscoped block under .k-code-block?

@kaiarrowood I’m not sure that’s appropriate to do as part of this PR.

Generally, components in Kongponents use scoped style blocks so that’s what I’m sticking to by default. It’s just that some styles in KCodeBlock can’t be scoped for various reasons (e.g. when applying styles to elements slotted in from the outside).

However, I do actually think that a component library should generally **not** use scoped styles and rather rely solely on using properly prefixed class names (like we’re already doing). This practice has the advantage of all selectors having lower specificity and are therefore easier to override in a consuming application:

```css
/* specificity 0 2 0 */
.k-code-block[v-scoped-style-attribute]

/* specificity 0 1 0 */
.k-code-block
```

The reason why I would avoid doing this in this PR is that because of the change to the selectors’ specificity, this is a potentially breaking change: Selectors that previously had a specificity of "0 2 0" will now have "0 1 0" and depending on a consuming project’s stylesheet order, a CSS rule set with a selector like `.k-button` would suddenly start overriding rule set in this component with the same selector.

Reply:
(Granted, I've had to reread your comment a few times to make sure I understand what you're saying here - specificity is a complex thing. So apologies if I misunderstood AGAIN). Are you saying you don't want to increase the specificity of the unscoped styles by adding another selector in? ie. `.k-code-block .k-code-block-secondary-actions` is harder to override than `.k-code-block-secondary-actions`

Reply:
Oh, of course, that makes more sense now. I misunderstood you. I’ve made this change now.

---

> Are you saying you don't want to increase the specificity of the unscoped styles by adding another selector in? ie. .k-code-block .k-code-block-secondary-actions is harder to override than .k-code-block-secondary-actions

Yes, that’s more or less accurate, but is only really of value in a wider sense where it’s library standard.

Ideally, those two things are true:

- Selectors are written in such a way that conflicts are unlikely (i.e. prefixed)
- Component styles use unscoped style blocks

Then, you have both robust styles (sure, not quite as robust as with scoping) _and_ as low as possible specificity which makes it easier to override unwanted styles in a consuming project (while surely everyone would like to avoid the need for that, it just happens).

That being said, I certainly don’t want to propose this as part of this PR. 😁